public/js/app.js

```diff
@@ -0,0 +1,106 @@
+(function () {
+    'use strict';
+    // MakeBlob allow to format a real octet-stream
+    const makeblob = (dataURL) => {
+        var BASE64_MARKER = ';base64,';
+        if (dataURL.indexOf(BASE64_MARKER) == -1) {
+            var parts = dataURL.split(',');
+            var contentType = parts[0].split(':')[1];
+            var raw = decodeURIComponent(parts[1]);
+            return new Blob([raw], { type: contentType });
+        }
+        var parts = dataURL.split(BASE64_MARKER);
+        var contentType = parts[0].split(':')[1];
+        var raw = window.atob(parts[1]);
+        var rawLength = raw.length;
+
+        var uInt8Array = new Uint8Array(rawLength);
+
+        for (var i = 0; i < rawLength; ++i) {
+            uInt8Array[i] = raw.charCodeAt(i);
+        }
+
+        return new Blob([uInt8Array], { type: contentType });
+    };
+    // Call the backend to send the picture to Azure
+    const sendImage = (imagePath) => {
+        let img = makeblob(imagePath)
+        $.ajax({
+            // NOTE: You must use the same location in your REST call as you used to obtain your subscription keys.
+            //   For example, if you obtained your subscription keys from westcentralus, replace "westus" in the 
+            //   URL below with "westcentralus".
+            url: "/sendimage",
+            // beforeSend: function(xhrObj){
+            //     // Request headers, also supports "application/octet-stream"
+            //     xhrObj.setRequestHeader("Content-Type","application/octet-stream");
+            // },
+            contentType: 'application/octet-stream',
+            type: "POST",
+            processData: false,
+            // Request body
+            data: img,
+            success: function (data) {
+                // Get face rectangle dimensions
+
+                var faceRectangle = data[0].faceRectangle;
```

Comment:
We need to add some defensive code here. Sometimes data could equal [] (if the API does not succeed to analyse the incoming data)